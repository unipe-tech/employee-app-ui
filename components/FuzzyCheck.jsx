/* eslint-disable react-hooks/rules-of-hooks */
import { useEffect } from "react";
import { Alert } from "react-native";
import { useDispatch,useSelector } from "react-redux";

import { setMistmatch as setBankMismatch } from "../store/slices/bankSlice";
import { setMistmatch as setPanMistach } from "../store/slices/panSlice";

const fuzz = require("fuzzball");

const FuzzyCheck = ({step}) => {
  const name = useSelector((state) => state.aadhaar.data?.name);
  const dispatch = useDispatch();
  const checkName =
    step === "PAN"
      
      ? useSelector((state) => state.pan.data?.name)
      : useSelector((state) => state.bank.data?.accountHolderName);
  const score = 100 - fuzz.token_set_ratio(name, checkName);
  useEffect(() => {
    console.log("FuzzyCheck", step, score);
    if (step === "PAN") {
      dispatch(setPanMistach(score));
    } else {
      dispatch(setBankMismatch(score));
    }
  }, [score]);

if(score > 20){
  return Alert.alert("Name mismatch",
  `Your Aadhaar Card name ${name} does not match with your ${step} name ${checkName}`)
}  
return null;
};

export default FuzzyCheck;
