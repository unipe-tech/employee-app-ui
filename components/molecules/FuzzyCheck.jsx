import React, { useEffect } from "react";
import { Alert } from "react-native";
import { useSelector, useDispatch } from "react-redux";
import { setMismatch as setBankMismatch } from "../../store/slices/bankSlice";
import { setMismatch as setPanMismatch } from "../../store/slices/panSlice";
const fuzz = require("fuzzball");

export const MismatchScore = ({ string, checkString }) => {
  const score = 100 - fuzz.token_set_ratio(string, checkString);
  return score;
};

const FuzzyCheck = (props) => {
  const dispatch = useDispatch();
  const aadhaarName = useSelector((state) => state.aadhaar.data?.name);
  const score = MismatchScore({ string: props.name, checkString: aadhaarName});
  useEffect(() => {
    console.log("FuzzyCheck", props.step, score);
    if (props.step == "PAN") {
      dispatch(setPanMismatch(score));
    } else {
      dispatch(setBankMismatch(score));
    }
  }, [score]);

  return (
    <>
      {score > 20
        ? Alert.alert(
            "Name mismatch",
            `Your Aadhaar Card name ${aadhaarName} does not match with your ${props.step} name ${checkName}`
          )
        : null}
    </>
  );
};

export default FuzzyCheck;
