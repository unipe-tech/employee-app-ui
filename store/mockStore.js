const store = {
  // Auth

  auth: {
    id: 1,
    phoneNumber: "9876543210",
    verifyStatus: "PENDING",
  },

  // Aadhaar

  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Xyz garg",
      },
    },
    number: "789456123010",
    submitOTPtxnId: "987",
    verifyMsg: "This otp has been sent",
    verifyStatus: "PENDING",
  },

  // Bank

  bank: {
    accountNumber: "157894561230",
    ifsc: "INDB0000123",
    upi: "9876543210@upi",
    accountHolderName: "Xyz Goyal",
    verifyStatus: "PENDING",
    verifyMsg: "The verification is pending",
    bankName: "IndusInd Bank",
    bankBranch: "Mumbai",
    branchCity: "Mumbai",
  },

  // ESIC

  esic: {
    portal: { estCode: "", ipNumber: "" },
    familyDetails: {
      fatherHusband: { relation: "", name: "" },
      nominee: { relation: "", name: "" },
    },
    address: {
      present: { street: "", state: "", district: "", pincode: "" },
      permanent: { street: "", state: "", district: "", pincode: "" },
      nominee: { street: "", state: "", district: "", pincode: "" },
    },
  },

  // License

  license: {
    number: "",
    name: "",
    dob: "",
    bloodGroup: "",
    classes: {},
    rto: {},
    validity: {},
    verifyMsg: "",
    verifyStatus: "PENDING",
  },

  // Navigation

  navigation: { currentScreen: "Welcome" },

  // PAN

  pan: {
    dob: "02/02/2002",
    email: "xyz@unipe.co",
    gender: "Male",
    name: "Xyz Goyal",
    number: "ABCDE1234F",
    verifyMsg: "Verification Pending",
    verifyStatus: "PENDING",
  },

  // Profile

  profile: {
    maritalStatus: "Married",
    educationalQualification: "12th pass",
    alternatePhone: "+913216549870",
    email: "abc@unipe.co",
    selfie: "",
  },

  // Timer

  timer: { login: 2 * 60, aadhaar: 10 * 60 },
};

export default store;
