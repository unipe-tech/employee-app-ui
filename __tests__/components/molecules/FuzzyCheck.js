import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import FuzzyCheck from "../../../components/molecules/FuzzyCheck";
import configureStore from "redux-mock-store";
import mockStore from "../../../store/mockStore";
import { NavigationContainer } from "@react-navigation/native";
import { Provider, useSelector } from "react-redux";

const middleware = [];

const mockedStore = configureStore(middleware);

const createdStore = mockedStore(mockStore);

describe("FuzzyCheck", () => {
  test("renders UI correctly", () => {
    const tree = render(
      <Provider store={createdStore}>
        <NavigationContainer>
          <FuzzyCheck />
        </NavigationContainer>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
