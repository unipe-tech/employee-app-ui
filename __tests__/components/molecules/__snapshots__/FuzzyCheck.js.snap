// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuzzyCheck renders UI correctly 1`] = `null`;
